### ISSUE TITLE
Array casting from channelview question

### ISSUE THREAD

**Issue by davidbp** [JuliaImages/juliaimages.github.io#60]
Hello,

I am not sure about how to get an Array containing the values resulting from a convolution to an image.

This is what I tried:
```
using Images
using TestImages
img = testimage("lena");
K =  [[1,1,1]  [0,0,0]  [-1,-1,-1]];
kernel = centered(K)
border_decision = Inner()
result_conv     = imfilter(img, kernel, border_decision)
result_conv_arr = size(Array(channelview(result_conv)))
```
prints
```
BoundsError: attempt to access 3×254×254 Array{Float32,3} at index [Base.IdentityUnitRange(1:3), Base.IdentityUnitRange(2:255), Base.IdentityUnitRange(2:255)]
```
Which I don't quite understand.

Nevertheless, when working with `Pad()` (or nothing which defaults to Pad) then the API behaves as I would expect:
```
border_decision = Pad()
result_conv     = imfilter(img, kernel, border_decision)
result_conv_arr = size(Array(channelview(result_conv)))
```
Prints
```
(3, 256, 256)
```

- How can I get an Array in the first case?

- Is there a method to retrieve the Array containing the data that works no matter how the offsetarray is?

**Comment by davidbp** [comment]
I´ve seen I can do this

```
Array(reshape(channelview(result_conv),(3, 254, 254)))
```
Which returns exactly what I wanted
```
3×254×254 Array{Float32,3}
```
I think it is too complicated and prone to errors.  I´ve inserted `(3, 254, 254)` manually. Is there an straightforward way to do this that I missed?

If not I can work on a function to do this because it´s very useful to convert an image to an Array for various reasons (working with other packages, comapring results with other image frameworks etc)

**Comment by johnnychen94** [comment]
Hi, you can try `collect` or `float.` instead of `Array` here.

Use `Array` to do conversion is prone to error for arrays that are not 1-based. E.g., `Array(centered(rand(3, 3)))` raises the same error here.

**Comment by davidbp** [comment]
Thanks @johnnychen94 , maybe adding an example to the documentation would help future users.

**Comment by johnnychen94** [comment]
@davidbp Yes, and I deliberately keep this issue open until I add this to documentation :D

**Comment by timholy** [comment]
@davidbp, it's worth asking why you need an `Array`. Have you read [this doc section](https://juliaimages.org/latest/indexing/#Keeping-track-of-location-with-unconventional-indices-1)? And/or [this blog post](https://julialang.org/blog/2017/04/offset-arrays)? 

The point is, the existing behavior:
```julia
julia> summary(result_conv)
"254×254 OffsetArray(::Array{RGB{Float32},2}, 2:255, 2:255) with eltype RGB{Float32} with indices 2:255×2:255"
```
tells you far more about the meaning of the result than if we returned a 254×254 Array. You're trying to make Julia throw away a lot of useful stuff simply because Matlab/Python can't handle stuff like that.

**Comment by davidbp** [comment]
Hello Tim, 

About
`You're trying to make Julia throw away a lot of useful stuff simply because Matlab/Python can't handle stuff like that.`

You are kind of right, but sometimes, I might want to use BLAS or other code that expects arrays of float32/float16 values. This kind of conversion is relevant and used, for example, in most flux tutorials.

**Comment by timholy** [comment]
Fair enough. But of course the *right* answer is to rewrite BLAS in Julia (which might actually happen some day, the benchmarks are starting to look rather encouraging), so that one gets the power of Julia's abstraction without sacrificing the performance of BLAS.

Meanwhile, it's a very simple operation to discard the unconventional axes (hint: `parent`), so do read that blog post.

**Comment by johnnychen94** [comment]
As a complementary:

It's not always right to call `parent`. For example, when the real data is hidden two layers deep:

```julia
julia> using ImageFiltering

julia> x = rand(3, 3)
3×3 Array{Float64,2}:
 0.334823  0.167111   0.35506 
 0.338576  0.865595   0.872704
 0.694813  0.0551154  0.014521

julia> centered(x)
3×3 OffsetArray(::Array{Float64,2}, -1:1, -1:1) with eltype Float64 with indices -1:1×-1:1:
 0.334823  0.167111   0.35506 
 0.338576  0.865595   0.872704
 0.694813  0.0551154  0.014521

julia> parent(centered(x))
3×3 Array{Float64,2}:
 0.334823  0.167111   0.35506 
 0.338576  0.865595   0.872704
 0.694813  0.0551154  0.014521

julia> parent(centered(centered(x)))
3×3 OffsetArray(::Array{Float64,2}, -1:1, -1:1) with eltype Float64 with indices -1:1×-1:1:
 0.334823  0.167111   0.35506 
 0.338576  0.865595   0.872704
 0.694813  0.0551154  0.014521
```

**Comment by timholy** [comment]
Oof, good catch. I thought we'd fixed that in https://github.com/JuliaArrays/OffsetArrays.jl/pull/67, but I guess we didn't cover enough cases.

**Comment by johnnychen94** [comment]
I think returning the _direct_ parent array is the expected behavior of `parent` and there's no need to fix. For example, `parent(::SubArray)` and `parent(::MappedArray)` work the same.
